### DISCUSSION TITLE
Modal Persistence Issue During Route Navigation from Intercepting Routes

### DISCUSSION THREAD

**Discussion by ankitmukhia** [vercel/next.js#70699] [App Router]
I am experiencing a challenge with modals in my Next.js application that utilize intercepting routes. When I navigate from the /works page, where a modal is opened, to the /projects page, the modal does not unmount or close as expected. Instead, it remains visible, which i cannot find way around ?

**Comment by suprunchuk** [comment]
**1.  Use a global state management solution:**

If you're using a state management library like Redux or Zustand, you can store the modal's open/closed state globally and update it when navigating between pages. Here's an example using Zustand:

```tsx
#modalStore.ts

import create from 'zustand';

interface ModalState {
  isOpen: boolean;
  setIsOpen: (isOpen: boolean) => void;
}

export const useModalStore = create<ModalState>((set) => ({
  isOpen: false,
  setIsOpen: (isOpen) => set({ isOpen }),
}));
```
Then in your components:

```tsx
#page.tsx

import { useModalStore } from '@/store/modalStore';
import { useEffect } from 'react';

const WorksPage = () => {
  const { isOpen, setIsOpen } = useModalStore();

  useEffect(() => {
    return () => {
      setIsOpen(false);
    };
  }, []);

  // Rest of your component code
};
```


**2. Use Next.js 13's new [usePathname ](https://nextjs.org/docs/app/api-reference/functions/use-pathname) hook:**

If you're using Next.js 13 or later with the App Router, you can use the usePathname hook to detect route changes:

```tsx
#page.tsx
import { usePathname } from 'next/navigation';
import { useEffect, useState } from 'react';

const WorksPage = () => {
  const pathname = usePathname();
  const [isModalOpen, setIsModalOpen] = useState(false);

  useEffect(() => {
    setIsModalOpen(false);
  }, [pathname]);

  // Rest of your component code
};

```
